cooperator/models/company.py

```diff
@@ -100,21 +100,61 @@ def _compute_base_logo(self):
         translate=True,
         help="Text to display aside the checkbox to approve the generic rules.",
     )
+    cooperator_certificate_mail_template = fields.Many2one(
+        comodel_name="mail.template",
+        string="Certificate email template",
+        domain="[('model', '=', 'res.partner')]",
```

Comment:
Maybe we can assign by default the mail template of cooperator, so we don't need to select it every time we create a new company.
Also, if is not assigned, it gets the cooperator template? Should be this field required?

Reply:
Agree with @DaniilDigtyar.
Also: perhaps the domain should add the ('is_cooperator_template', '=', True) element (in addition to the correspondant model one)

Reply:
I see now that is assigned when creating the company.

Reply:
I'm not entirely sure what the proposal is. This is the philosophy of the current state of the PR:

There exists a template (in truth, several, but let's simplify to 1) defined in the XML data. Let's call this the global template. This template is not itself used anywhere. Instead, when a company is created, a copy of the global template is created and set on the field in res.company.

The upside of this approach is that none of the non-global templates interfere with each other. The downside is that you get the following slight UX problem: When creating a company, you leave the mail template field empty. After clicking on 'save', the copy is created and the field is populated. If you now _erase_ the contents of that field, nothing will stop you, and you will break the `cooperator` module.

I have bypassed one of the UX problems by making the mail template field invisible until after clicking on 'save'. This means the user won't be confused by an empty field suddenly being populated, or feeling like they have to populate these fields during creation when in fact they needn't.

The other problem I am not certain how to work around. The field can't be mandatory, because it's empty during creation. Similar problem in re adding a constraint. Maybe you could do an onchange or write that throws an error; I'm not sure.

---

Your proposal, as I understand it, is to use the global by default on companies, either by making it the default value of the field, or by making it a fallback value in a helper getter method (both are fine; I'd probably prefer the former over the latter, but it's just about the same). This fixes the UX problems (although we may need to add a help text that 'empty' is a valid value for the template field), but does change the philosophy: where previously all templates were always separate, they are now identical-unless-declared-otherwise.

I'm fine with that change in philosophy, but it has the downside that a user at one company may change the template without realising that this affects all other companies.

@DaniilDigtyar @XavierBonetViura Let me know which approach you prefer.

Reply:
No, you are right and this is correct implementation. It was a misunderstanding about how it was working because we didn't understand the assignation of the relations after the creation of the company record.

Reply:
This philosophy of your implementation is the one that we need in Som Comunitats projects, because we need to have different `mail.templates` for all companies, even if there are little variation between them.

But maybe for a more global need of `cooperator`, it will make sense to use the global mail template as the default for all the companies and not copy them on every creation, because this will cause an extensive list of same `mail.template` shared between all the companies (because there isn't a `company_id` by default)

Reply:
Creating systematically a .copy() of each email_template and assign those new templates to every new company that we create is an aproach the have some issues:

- We want to have the possibility to have it, ... but **sure that not all companies will need to have it** (personalized templates). Perhaps only FEW companies will want to have it.
- Caution with the templates Translation effort. I we build new templates for absolutelly each new company we will need to proceed to translate it per each company. I can be crazy to maintain.
- If we proceed to do CHANGES into the 'cooperator' templates by code (adding globally new fuctionallity),.... those improvements will not apply at all at any company because we have ALL the templates personalized.

Perhaps it will be better to have another aproach:
1) have the new fields available at res_company to select personalized templates,... but allow to have it EMPTY (by default). 
2) If those fields are EMPLY, the code will use the 'corporate' (cooperator) base templates.
3) If a company need to PERSONALYZE A TEMPLATE, allow a button at mail_template model to do a COPY of any of the coopetaror templates,... and assign company_id field to those new copied templates.
4) filally at the res_company fields that point to those persolalyzed email_templates, force the domain in order to ONLY show to be selected the templates that match the field company_id (and that are cooperator ones)

Reply:
@XavierBonetViura I have implemented the above proposal, where empty = default. Thank you for that proposal.

I have not presently implemented a 'copy' button. Let's come back to that later, maybe.